code/datums/religions.dm

```diff
@@ -25,6 +29,80 @@
 	return TRUE // Nothing to see here, but redefined in some other religions !
 
+/* ---- RELIGIOUS CONVERSION ----
+ * convertAct() -> convertCeremony() -> convertCheck() -> convert()
+ * Redefine 'convertCeremony' to play out your snowflake ceremony/interactions in your religion datum.
+ * In a saner language, convertCeremony() and convertCheck() would be private methods. Those are UNSAFE procs. Call convertAct() instead.
+ */
+
+/* ConvertAct() : here we check if eveything is in place for the conversion, and provide feedback if needed. Sanity for the preacher or the target belongs to the verb in the bible.
+ * - preacher : the guy doing the converting
+ * - subject : the guy being converted
+ * - B : the bible using for the conversion
+ */
+/datum/religion/proc/convertAct(var/mob/living/preacher, var/mob/living/subject, var/obj/item/weapon/storage/bible/B)
+	if (B.my_rel != src) // BLASPHEMY
+		to_chat(preacher, "<span class='warning'>You are a heathen to this God. You feel [B.my_rel.deity_name]'s wrath strike you for this blasphemy.</span>")
+		preacher.fire_stacks += 5
+		preacher.IgniteMob()
+		preacher.emote("scream",,, 1)
+		return FALSE
+	if (preacher != religiousLeader.current)
+		to_chat(preacher, "<span class='warning'>You fail to muster enough mental strength to begin the conversion. Only the Spiritual Guide of [name] can perfom this.</span>")
+		return FALSE
+	if (subject.mind.faith == src)
+		to_chat(preacher, "<span class='warning'>You and your target follow the same faith.</span>")
+		return FALSE
+	else
+		return convertCeremony(preacher, subject)
+
+/* ConvertCeremony() : the RP ceremony to convert the newfound person.
+ Here we check if we have the tools to convert and play out the little interactions. */
+
+ // This is the default ceremony, for Christianity/Space Jesus
+/datum/religion/proc/convertCeremony(var/mob/living/preacher, var/mob/living/subject)
+	var/held_beaker = preacher.find_held_item_by_type(/obj/item/weapon/reagent_containers)
+	var/obj/item/weapon/reagent_containers/B = preacher.held_items[held_beaker]
+	if (!held_beaker || B.reagents.get_master_reagent_name() != "Holy Water")
+		to_chat(preacher, "You need to hold Holy Water to begin to conversion.")
+		return FALSE
+	subject.visible_message("\the [preacher] attemps to convert the [subject] to [name].")
+	if(!convertCheck(subject))
+		subject.visible_message("\The [subject] refuses conversion.")
+		return FALSE
+
+	// Everything is ok : begin the conversion
+	splash_sub(B.reagents, subject, 5, preacher)
+	subject.visible_message("\The [subject] is blessed by \the [preacher] and embraces [name]. Praise [deity_name]!")
+	convert(subject, preacher)
+	return TRUE
+
+// Here we check if the subject is willing
+/datum/religion/proc/convertCheck(var/mob/living/subject)
+	var/choice = input(subject, "Do you wish to become a follower of [name]?","Religious converting") in list("Yes", "No")
+	return (choice == "Yes" ? TRUE : FALSE)
```

Comment:
Isn't this just `return choice == Yes`?

Reply:
It works as a ternary operator
![converting](https://user-images.githubusercontent.com/31417754/33517966-a1eefc9c-d78d-11e7-9216-2ad004b856e8.png)

Reply:
Isn't this just `return choice == Yes`?

Reply:
You're right